plugin-locationcomponent/src/test/java/com/mapbox/maps/plugin/locationcomponent/LocationComponentPluginImplTest.kt

```diff
@@ -0,0 +1,354 @@
+  }
+
+  @Test
+  fun testOnPuckBearingAnimatorDefaultOptionsUpdated() {
+    val valueAnimator = mockk<ValueAnimator.() -> Unit>(relaxed = true)
+    preparePluginInitialisationWithEnabled()
+    locationComponentPlugin.onPuckBearingAnimatorDefaultOptionsUpdated(valueAnimator)
+    verify { locationPuckManager.updateBearingAnimator(valueAnimator) }
+  }
+
+  @Test
+  fun testApplySettingsEnabled() {
+    preparePluginInitialisationWithDisabled()
+    every { locationPuckManager.isLayerInitialised() } returns false
+    locationComponentPlugin.enabled = true
+    verify { locationPuckManager.updateSettings(any()) }
+  }
+
+  @Test
+  fun testApplySettingsDisabled() {
+    preparePluginInitialisationWithEnabled()
+    locationComponentPlugin.enabled = false
+    verify(exactly = 0) { locationPuckManager.updateSettings(any()) }
+    verify { locationPuckManager.cleanUp() }
+    verify { locationProvider.unRegisterLocationConsumer(any()) }
+    assertFalse(locationComponentPlugin.isLocationComponentActivated)
+    assertNull(locationComponentPlugin.locationPuckManager)
+  }
+
+  @Test
+  fun testLocationProviderRegisterStopStart() {
+    preparePluginInitialisationWithEnabled()
+
+    verify(exactly = 1) { locationProvider.registerLocationConsumer(any()) }
+    verify(exactly = 0) { locationProvider.unRegisterLocationConsumer(any()) }
+    locationComponentPlugin.onStop()
+    verify(exactly = 1) { locationProvider.registerLocationConsumer(any()) }
+    verify(exactly = 1) { locationProvider.unRegisterLocationConsumer(any()) }
+
+    every { delegateProvider.getStyle(capture(styleCallbackSlot)) } returns Unit
+    locationComponentPlugin.onStart()
+    if (styleCallbackSlot.isCaptured) {
+      styleCallbackSlot.captured.invoke(style)
+    }
+    verify(exactly = 2) { locationProvider.registerLocationConsumer(any()) }
+    verify(exactly = 1) { locationProvider.unRegisterLocationConsumer(any()) }
+  }
+
+  @Test
+  fun testLocationProviderRegisterDisableEnable() {
+    every { style.addStyleLayer(any(), any()) } returns ExpectedFactory.createValue()
+
+    preparePluginInitialisationWithEnabled()
+
+    verify(exactly = 1) { locationProvider.registerLocationConsumer(any()) }
+    verify(exactly = 0) { locationProvider.unRegisterLocationConsumer(any()) }
+    locationComponentPlugin.enabled = false
+    verify(exactly = 1) { locationProvider.registerLocationConsumer(any()) }
+    verify(exactly = 1) { locationProvider.unRegisterLocationConsumer(any()) }
+
+    every { delegateProvider.getStyle(capture(styleCallbackSlot)) } returns Unit
+    locationComponentPlugin.enabled = true
+    if (styleCallbackSlot.isCaptured) {
+      styleCallbackSlot.captured.invoke(style)
+    }
+    verify(exactly = 1) { style.addStyleLayer(any(), any()) }
+    verify(exactly = 2) { locationProvider.registerLocationConsumer(any()) }
+    verify(exactly = 1) { locationProvider.unRegisterLocationConsumer(any()) }
+  }
+}
```

Comment:
It's pretty hard to check if all cases are covered but can we have a test (or did I miss it) to check that recently added condition https://github.com/mapbox/mapbox-maps-android/blob/c62398a7ea11e0c5ac99dde69962baaa12b78827/plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/LocationComponentPluginImpl.kt#L133

Reply:
It's covered in `testOnStart`, `testOnStartWithPuckDisabled` and `testOnStartWithPuckAlreadyInitialisedAndPluginActivated`.